data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/RemotePeerForwarder.java

```diff
@@ -177,6 +190,23 @@ private boolean isAddressDefinedLocally(final String address) {
         }
     }
 
+    private List<Record<Event>> batchRecordsForForwarding(final String destinationIp, final List<Record<Event>> records) {
+        if (!peerBatchingQueueMap.containsKey(destinationIp)) {
+            peerBatchingQueueMap.put(destinationIp, new LinkedBlockingQueue<>(forwardingBatchSize * 5));
```

Comment:
In [this comment](https://github.com/opensearch-project/data-prepper/pull/2175#discussion_r1084743826), I asked about what would happen when requests to the clients are running behind and the processing threads are just adding more async requests to the thread pool.

Is this 5 multiplier a way to provide this type of back-pressure? I think this could make sense. I do wonder if the number of worker threads could also be part of this multiplier.

```
forwardingBatchSize * workerThreads * BATCH_QUEUE_DEPTH
```

Reply:
5 was an arbitrary number to give some overhead for multiple workers writing in sync before the drainTo occurs or a high batch_size with a low forwarding_batch_size. 

I like the formula you proposed. My only concern is memory usage, but that shouldn't be too big of an issue with the heap circuit breaker